test/open_telemetry_test.exs

```diff
@@ -4,13 +4,40 @@ defmodule OpenTelemetryTest do
   require OpenTelemetry.Tracer, as: Tracer
   require OpenTelemetry.Span, as: Span
 
+  require Record
+  @fields Record.extract(:span_ctx, from_lib: "opentelemetry_api/include/opentelemetry.hrl")
+  Record.defrecordp(:span_ctx, @fields)
+
+  @fields Record.extract(:link, from_lib: "opentelemetry_api/include/opentelemetry.hrl")
+  Record.defrecordp(:link, @fields)
```

Comment:
I had to do something similar on my tests, would it make sense to create a issue about putting this logic inside the `.ex` files? Like what @garthk sent on Slack: https://github.com/opencensus-beam/opencensus_elixir/blob/master/lib/opencensus/span.ex

Reply:
Unsure. The problem is keeping it from appearing that anyone besides those using tests should be using them.

My fear is if we put it and maybe even add structs like in OpenCensus Elixir in the main Otel module it will lead to users to do things they shouldn't :)

I'd rather make even the records "private" but that isn't an option because third party processors and exporters need them.

Reply:
Your reasoning is about making these records opaque to users, so they don't directly match and manipulate them?

This sounds good to me, tbh, but to make the case that we talked about yesterday work (about linking Oban's workers span with the span of whoever created the job), I had to match the record directly.

Here is the snippet that I had to do this:
```elixir
        defp add_otel_span_link({_, _, :undefined, _, _, _, _, _}, args), do: args                                                                                                                                                           
        defp add_otel_span_link({_, :undefined, _, _, _, _, _, _}, args), do: args                                                                                                                                                           
                                                                                                                                                                                                                                             
        defp add_otel_span_link({_, trace_id, span_id, _, _, _, _, _}, args) do                                                                                                                                                              
          Map.put(args, "otel_span_link", %{"trace_id" => trace_id, "span_id" => span_id})                                                                                                                                                   
        end                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                             
        defp add_otel_span_link(_, args), do: args  ```

In this case I couldn't find a way to properly pass the link across this boundary. Is there anything in the library, or from the specification, that would help with this case? If there isn't, users will have no other option for now (I believe), until this issue can be properly discussed and best practices set

Reply:
Isn't that a span context you are matching the trace id and span id out of there? You can just pass that to the link function now.

There may be cases that we didn't cover that result in code like this but we need to fix that instead of just exporting the records.

Reply:
@tsloughter that `args` will be serialized and stored into PostgreSQL, so I can't pass directly `span_ctx` as it's not json serializable

Reply:
I have been thinking if maybe this is a use case for https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md but I haven't yet understood how to use it in thise library

Reply:
Ah yes, it is a propagator. A similar one would be one that serializes the span ctx to kafka headers that I need to work on. But this reminded me that we actually have functions in `Span` module for this:

https://github.com/open-telemetry/opentelemetry-erlang-api/blob/master/lib/open_telemetry/span.ex#L30-L46

Reply:
So for the time being, instead of dealing with a propagator, best to just use the `Span.span_id`/`Span.trace_id` functions.